File: app/models/asset.py
from pydantic import BaseModel
from typing import Optional

class AssetCreate(BaseModel):
    name: str
    site: str
    status: str

class AssetResponse(BaseModel):
    id: int
    name: str
    site: str
    status: str

class AssetUpdate(BaseModel):
    name: Optional[str] = None
    site: Optional[str] = None
    status: Optional[str] = None

File: frontend/src/types/asset.ts
export interface Asset {
    id: number,
    name: string,
    site: string,
    status: string
}

export interface AssetResponse {
    success: boolean,
    data: Asset
}

export interface PaginatedResponse<T> {
    results: T[],
    count: number,
    page: number,
    page_size: number,
    prev: string | null,
    next: string | null
}